tests/route/test_route_perf.py

```diff
@@ -0,0 +1,117 @@
+ROUTE_TABLE_NAME = 'ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY'
+
+def prepare_dut(duthost):
+    # Set up interface
+    duthost.shell('sudo config interface ip add Ethernet72 10.1.0.54/31')
+    
+    # Set up neighbor
+    duthost.shell('sudo ip neigh add 10.1.0.37 lladdr 55:54:00:ad:48:98 dev Ethernet72')
+
+def cleanup_dut(duthost):
+    # Delete neighbor
+    duthost.shell('sudo ip neigh del 10.1.0.37 dev Ethernet72')
+
+    # remove files
+    duthost.shell('rm {}'.format(ROUTE_JSON))
+    duthost.shell('docker exec -i swss rm {}'.format(ROUTE_JSON))
+
+    # remove interface
+    duthost.shell('sudo config interface ip remove Ethernet72 10.1.0.54/31')
+
+def generate_route_file(duthost, prefixes, op):
+    route_data = []
+    for prefix in prefixes:
+        key = 'ROUTE_TABLE:' + prefix
+        route = {}
+        route['ifname'] = 'Ethernet72'
+        route['nexthop'] = '10.1.0.37'
+        route_command = {}
+        route_command[key] = route
+        route_command['OP'] = op
+        route_data.append(route_command)
+
+    # Copy json file to DUT
+    duthost.copy(content=json.dumps(route_data, indent=4), dest=ROUTE_JSON)
+
+def exec_routes(duthost, num_routes, op):
+    # generate ip prefixes of routes
+    prefixes = ['%d.%d.%d.%d/%d' % (101 + int(idx_route / 256 ** 2), int(idx_route / 256) % 256, idx_route % 256, 0, 24)
+                for idx_route in range(num_routes)]
+
+    # Generate json file for routes
+    generate_route_file(duthost, prefixes, op)
+
+    # Copy route file to swss container
+    duthost.shell('docker cp {} swss:/'.format(ROUTE_JSON))
+    
+    # Check the number of routes in ASIC_DB
+    start_num_route = int(duthost.shell('sonic-db-cli ASIC_DB keys \'{}*\' | wc -l'.format(ROUTE_TABLE_NAME))['stdout'])
+    
+    # Calculate expected number of route and record start time
+    if op == 'SET':
+        expected_num_routes = start_num_route + num_routes
+    elif op == 'DEL':
+        expected_num_routes = start_num_route - num_routes
+    else:
+        pytest.fail('Operation {} not supported'.format(op))
+    start_time = datetime.now()
+
+    # Apply routes with swssconfig
+    duthost.shell('docker exec -i swss swssconfig /{}'.format(ROUTE_JSON))
+
+    # Wait until the routes set/del applys to ASIC_DB
+    def _check_num_routes(expected_num_routes):
+        # Check the number of routes in ASIC_DB
+        num_routes = int(duthost.shell('sonic-db-cli ASIC_DB keys \'{}*\' | wc -l'.format(ROUTE_TABLE_NAME))['stdout'])
+        return num_routes == expected_num_routes
+    if not wait_until(ROUTE_TIMEOUT, 0.5, _check_num_routes, expected_num_routes):
+        pytest.fail('failed to add routes within time limit')
+
+    # Record time when all routes show up in ASIC_DB
+    end_time = datetime.now()
+
+    # Check route entries are correct
+    asic_route_keys = duthost.shell('sonic-db-cli ASIC_DB keys \'{}*\''.format(ROUTE_TABLE_NAME))['stdout_lines']
```

Comment:
>duthost.shell [](start = 22, length = 13)

Is there already a function for this?

Reply:
I did not find a function that does this job. I did find a few other places that use similar `sonic-db-cli` and `redis-cli` to get keys.

Reply:
I am worried about the performance. I found a better solution [https://stackoverflow.com/a/18247870/2514803](https://stackoverflow.com/a/18247870/2514803)

---
In reply to: [505121217](https://github.com/Azure/sonic-mgmt/pull/2344#discussion_r505121217) [](ancestors = 505121217)